core-examples/rust/message-monitoring/src/main.rs

```diff
@@ -0,0 +1,245 @@
+use clap::{CommandFactory, Parser};
+use serde_json::{from_value, json};
+use std::{
+    env,
+    sync::Arc,
+    time::{SystemTime, UNIX_EPOCH},
+};
+use tokio::time::{sleep, Duration};
+use ton_client::{
+    abi::{encode_message, Abi, AbiContract, CallSet, ParamsOfEncodeMessage, Signer},
+    crypto::{nacl_sign_keypair_from_secret_key, KeyPair, ParamsOfNaclSignKeyPairFromSecret},
+    net::NetworkConfig,
+    processing::{
+        fetch_next_monitor_results, get_monitor_info, send_messages, MessageSendingParams, MonitorFetchWait, ParamsOfFetchNextMonitorResults,
+        ParamsOfGetMonitorInfo, ParamsOfSendMessages,
+    },
+    ClientConfig, ClientContext,
+};
+
+#[derive(Parser, Debug)]
+#[clap(version, about, long_about = None)]
+struct Args {
+    #[clap(long, short, action)]
+    loop_nowait: bool,
+    #[clap(long, short, action)]
+    once_await: bool,
+    #[clap(long, short, action)]
+    await_all: bool,
+    #[clap(long, short, default_value_t = 10)]
+    message_count: u32,
+    #[clap(long, short, default_value_t = 3)]
+    wait_seconds: u32,
+}
+
+const LOOP_SLEEP_IN_SECONDS: u64 = 1;
+const QUEUE: &str = "Foo";
+
+#[tokio::main]
+async fn main() {
+    let args = Args::parse();
+    if !args.await_all && !args.loop_nowait && !args.once_await {
+        let mut cmd = Args::command();
+        cmd.print_help().unwrap();
+        std::process::exit(64);
+    }
+    let giver_abi = json!({
+        "ABI version": 2,
+        "header": ["time", "expire"],
+        "functions": [
+            {
+                "name": "sendTransaction",
+                "inputs": [
+                    {
+                        "name": "dest",
+                        "type": "address",
+                    },
+                    {
+                        "name": "value",
+                        "type": "uint128",
+                    },
+                    {
+                        "name": "bounce",
+                        "type": "bool",
+                    },
+                ],
+                "outputs": [],
+            },
+        ],
+    });
+
+    let network = match env::var("TON_NETWORK_ADDRESS") {
+        Ok(val) => val,
+        Err(_) => "http://localhost:4000/graphql".to_owned(), // gateway endpoint
```

Comment:
Why endpoint port is 4000 ?  
By default local-node is started on port  80

`docker run -d --name local-node -e USER_AGREEMENT=yes -p80:80 tonlabs/local-node`

Reply:
The local-node can't provide message-monitor capabilities without stitching api service through the gateway.